pkg/bundle/bundle.go

```diff
@@ -59,10 +59,9 @@ type bundle struct {
 	// resources whose informer only caches metadata.
 	directClient client.Client
 
-	// lister makes requests to the informer cache. Beware that resources whose
-	// informer only caches metadata, will not return underlying data of that
-	// resource. Use client instead.
-	lister client.Reader
+	// sourceLister makes requests to the informer cache. All cached source resources
+	// are expected to be full objects in a single namespace (the TrustNamespace).
+	sourceLister client.Reader
```

Comment:
suggestion (non-blocking): I like the lister -> sourceLister change a lot, it makes a lot of sense to treat sources and targets differently.

Does this mean that `directClient` would now only be used for interacting with targets? If so, might it be worth renaming it to `targetClient` or similar to make the name better match what the client is used for? Totally not needed though!